core/modules/filters/is.js

```diff
@@ -26,23 +26,41 @@ function getIsFilterOperators() {
 Export our filter function
 */
 exports.is = function(source,operator,options) {
-	// Dispatch to the correct isfilteroperator
-	var isFilterOperators = getIsFilterOperators();
-	if(operator.operand) {
-		var isFilterOperator = isFilterOperators[operator.operand];
-		if(isFilterOperator) {
-			return isFilterOperator(source,operator.prefix,options);
-		} else {
-			return [$tw.language.getString("Error/IsFilterOperator")];
-		}
-	} else {
+
+
+	if( !operator.operand) {
 		// Return all tiddlers if the operand is missing
 		var results = [];
 		source(function(tiddler,title) {
 			results.push(title);
 		});
 		return results;
 	}
+
+	// Get our isfilteroperators
+	var isFilterOperators = getIsFilterOperators(),
+	    subops = operator.operand.split("+"),
+		filteredResults = {},
+		results = [];
+	for (var t=0; t<subops.length; t++) {
+		var subop = isFilterOperators[subops[t]];
+		if(subop) {
+			filteredResults[subops[t]] = subop(source,operator.prefix,options);
+		} else {
+			return [$tw.language.getString("Error/IsFilterOperator")];
+		}
+		
+	}
+	
+    source(function(tiddler,title) {
```

Comment:
IMO the whole comparison here is wrong. ... The operand filters already returned "valid" values. .. Testing the again is just wasting time and also wrong.

Reply:
IMO The only thing which needs to be done here is 

 - have a look if there is only 1 operand
   - If it is, just return it --> fast lane
 - if there are 2 or more operands 
   - combine the results. because the operand functions only return valid titles.

Reply:
Ok, yeah, getting rid of the source call makes sense here.  But I'd still like to maintain the input order, unless that's affecting performance in a dramatic way.  And if we fast lane the single operand, the speed of multiple operands should matter less.

Reply:
The problem I have is, that the source test is wrong. The wrapper function is not allowed to test. The operand functions already return valid lists. The wrapper doesn't know how to test again. ...

Reply:
How does this look to you?  https://github.com/Jermolene/TiddlyWiki5/compare/master...mklauber:is-operator-multiple-categories?expand=1

Reply:
The wrapper here is just supposed to take the elements from the input, and figure out if they're in any of the outputs, and if so, add them to the output.